File: src/main/java/akostenko/aicars/model/CarModel.java
package akostenko.aicars.model;

import static java.lang.StrictMath.PI;
import static java.lang.StrictMath.toRadians;

public interface CarModel {
    /** <i>m</i> */
    double tyreRadius = 0.3;
    double frontTyreWidth = 0.25;
    double readTyreWidth = 0.35;
    /** height of center of mass, <i>m</i> */
    double massCenterHeight = 0.2;

    /** <i>kg</i> */
    double mass = 700;
    /** <i>kg*m</i> */
    double yawInertia = 500;
    /** <i>1/s</i> */
    double max_rpm = 12000;
    /** <i>1/s</i> */
    double min_rpm = 3500;
    /** non-dimensional */
    double cx = 0.8; // Cx
    /** non-dimensional */
    double cy = 1.0033; // Cy
    /** <i>m^2</i> */
    double frontArea = 1.1;
    /** <i>m^2</i> */
    double wingArea = 5;
    /** stiction coefficient between hot slick tyre and dry asphalt, non-dimensional */
    double tyreStiction = 1.3;
    /** rolling friction coefficient between rubber and asphalt, <i>m</i> */
    double tyreRollingFriction = 0.010;
    double axleTrack = 2; // m
    double wheelbase = 4.5; // m
    double massCenterOffset = .45; // from rear axle, measured as offset/wheelbase
    double length = 5.5; // m
    double maxSteering = 0.2*PI;
    double peakLateralForceAngle = toRadians(3);
}

File: game-engine/src/main/kotlin/akostenko/aicars/model/CarModel.kt
package akostenko.aicars.model

import org.apache.commons.math3.util.FastMath.PI
import org.apache.commons.math3.util.FastMath.toRadians

object CarModel {
    /** *m*  */
    val tyreRadius = 0.3
    val frontTyreWidth = 0.25
    val readTyreWidth = 0.35
    /** height of center of mass, *m*  */
    val massCenterHeight = 0.2

    /** *kg*  */
    val mass = 700.0
    /** *kg*m*  */
    val yawInertia = 500.0
    /** *1/s*  */
    val max_rpm = 12000.0
    /** *1/s*  */
    val min_rpm = 3500.0
    /** non-dimensional  */
    val cx = 0.8 // Cx
    /** non-dimensional  */
    val cy = 1.0033 // Cy
    /** *m^2*  */
    val frontArea = 1.1
    /** *m^2*  */
    val wingArea = 5.0
    /** stiction coefficient between hot slick tyre and dry asphalt, non-dimensional  */
    val tyreStiction = 1.3
    /** rolling friction coefficient between rubber and asphalt, *m*  */
    val tyreRollingFriction = 0.010
    val axleTrack = 2.0 // m
    val wheelbase = 4.5 // m
    val rearWeightPercent = .45
    val frontWeightPercent = 1 - rearWeightPercent
    val length = 5.5 // m
    val maxSteering = 0.2 * PI
    val peakLateralForceAngle = toRadians(5.0)
    /** starting from this speed (m/s) pure geometry turning physics is replaced with dynamic forces turning physics.  */
    val highSpeedCorneringThreshold = 10.0
}
